### PR TITLE
js[minor]: Add clonePublicDataset and listSharedExamples

### PR DESCRIPTION
**Pull request by bracesproul**

### CODE REVIEW

File: js/src/schemas.ts [hunk 1]
```diff
@@ -270,6 +270,8 @@ export interface Dataset extends BaseDataset {
   example_count?: number;
   session_count?: number;
   last_session_start_time?: number;
+  inputs_schema_definition?: KVMap;
+  outputs_schema_definition?: KVMap;
```
Comment (bracesproul) [inline js/src/schemas.ts:274]: cc @jakerachleff is this the proper interface to add these fields to? Or should they be on `BaseDataset`?

Comment (hinthornw) [reply js/src/schemas.ts:274]: It can be on the base i believe (that way in BE)

File: js/src/client.ts [hunk 2]
```diff
@@ -1528,6 +1528,53 @@ export class Client {
...
+        method: "GET",
+        headers: this.headers,
+        signal: AbortSignal.timeout(this.timeout_ms),
+        ...this.fetchOptions,
+      }
+    );
+    const result = await response.json();
+    if (!response.ok) {
+      throw new Error(
+        `Failed to list shared examples: ${response.status} ${response.statusText}`
```
Comment (hinthornw) [inline js/src/client.ts:1569]: Possible to include the full error message?

File: js/src/client.ts [hunk 3]
```diff
@@ -1528,6 +1528,53 @@ export class Client {
     return dataset as Dataset;
   }
 
+  /**
+   * Get shared examples.
+   * @param {string} shareToken The share token to get examples for.
```
Comment (hinthornw) [inline js/src/client.ts:1533]: Maybe say what a share token is / how to get it?

Comment (bracesproul) [reply js/src/client.ts:1533]: To confirm a share token is the UUID (or UUID w/ smith url) generated when you share something (e.g a run)?

Comment (hinthornw) [reply js/src/client.ts:1533]: Ya!

**Review by hinthornw — APPROVED** [review]

**Pull request merged**